### PR TITLE
Make debug port configurable

### PR DESCRIPTION
**Pull request by Flanker32**
Make debug port configurable, for issue #394

### CODE REVIEW

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/RunMojo.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
     
     public static final String FUNC_HOST_START_CMD = "func host start";
     public static final String FUNC_HOST_START_WITH_DEBUG_CMD = "func host start --language-worker -- " +
-            "\"-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005\"";
+            "\"-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=%d\"";
```
Comment (pragnagopa) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/RunMojo.java]: Debug options for java are specified with 
```
"-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=portNumber
```

This full string needs to be configurable not just the port as there might be other protocols that can be used for debugging.

File: azure-functions-maven-plugin/README.md [hunk 2]
```diff
@@ -109,10 +109,7 @@ Property | Required | Description
...
 `<httpProxyPort>` | false | Specifies an optional HTTP proxy port to connect to Azure through <br>The default value is **80**.
-`<localDebugAddress>` | false | Transport address for the connection. If server=n, attempt to attach to debugger application at this address. If server=y, listen for a connection at this address. The default value is **5005**.
-`<localDebugServer>` | false | If "y", listen for a debugger application to attach; otherwise, attach to the debugger application at the specified address. The default value is **"y"**.
-`<localDebugSuspend>` | false | If "y", VMStartEvent has a suspendPolicy of SUSPEND_ALL. If "n", VMStartEvent has a suspendPolicy of SUSPEND_NONE. The default value is **"n"**.
-`<localDebugTransport>` | false | Name of the transport to use in connecting to debugger application. It could be **dt_socket** or **dt_shmem** *(Windows Only)*. The default value is **"dt_socket"**. 
+`<localDebugConfig>` | false | The config string of debug options, you may visit [here](https://docs.oracle.com/javase/7/docs/technotes/guides/jpda/conninv.html) for more infos. The default value is `transport=dt_socket,server=y,suspend=n,address=5005`;
```
Comment (jdneo) [inline azure-functions-maven-plugin/README.md]: info -> information

We should avoid using abbreviations in all of the documents

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/RunMojo.java [hunk 3]
```diff
@@ -30,7 +30,7 @@
     
     public static final String FUNC_HOST_START_CMD = "func host start";
     public static final String FUNC_HOST_START_WITH_DEBUG_CMD = "func host start --language-worker -- " +
-            "\"-agentlib:jdwp=transport=%s,server=%s,suspend=%s,address=%s\"";
+            "\"-agentlib:jdwp=%s\"";
```
Comment (jdneo) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/RunMojo.java]: Why the string not start with `-agentlib....`

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 4]
```diff
@@ -140,6 +140,12 @@
     @Parameter(property = "httpProxyPort", defaultValue = "80")
     protected int httpProxyPort;
 
+    /**
+     * Config String for local debug
+     */
```
Comment (jdneo) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java]: Add `since xxx` xxx is the version number

**Pull request merged**